leshan-server-core/src/main/java/org/eclipse/leshan/server/security/oscore/OscoreSetting.java

```diff
@@ -0,0 +1,117 @@
+/*******************************************************************************
+ * Copyright (c) 2022    Sierra Wireless and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v20.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *     Sierra Wireless - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.leshan.server.security.oscore;
+
+import java.io.Serializable;
+import java.util.Arrays;
+
+import org.eclipse.leshan.core.util.Hex;
+
+public class OscoreSetting implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private final byte[] senderId;
+    private final byte[] recipientId;
+    private final byte[] masterSecret;
+    private final Integer aeadAlgorithm;
+    private final Integer hmacAlgorithm;
+    private final byte[] masterSalt;
+
+    public OscoreSetting(byte[] senderId, byte[] recipientId, byte[] masterSecret, Integer aeadAlgorithm,
+            Integer hmacAlgorithm, byte[] masterSalt) {
+        this.senderId = senderId;
+        this.recipientId = recipientId;
+        this.masterSecret = masterSecret;
+        this.aeadAlgorithm = aeadAlgorithm;
+        this.hmacAlgorithm = hmacAlgorithm;
+        this.masterSalt = masterSalt;
+    }
+
+    public byte[] getSenderId() {
+        return senderId;
+    }
+
+    public byte[] getRecipientId() {
+        return recipientId;
+    }
+
+    public byte[] getMasterSecret() {
+        return masterSecret;
+    }
+
+    public Integer getAeadAlgorithm() {
+        return aeadAlgorithm;
+    }
+
+    public Integer getHmacAlgorithm() {
+        return hmacAlgorithm;
+    }
```

Comment:
Maybe better to replace `Integer` by dedicated class/enum, like I did in : https://github.com/eclipse/leshan/pull/1204/files#diff-593f0c45e56ce01047c78d607542064f008dda0112d69f3c3097a237c5a0e32cR26